components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/websocket/InboundWebsocketResponseSender.java

```diff
@@ -53,10 +57,33 @@ public InboundWebsocketSourceHandler getSourceHandler() {
 
     @Override
     public void sendBack(MessageContext msgContext) {
+        String defaultContentType = sourceHandler.getDefaultContentType();
         if (msgContext != null) {
-
-            if (msgContext.getProperty(InboundWebsocketConstants.WEBSOCKET_SOURCE_HANDSHAKE_PRESENT) != null &&
-                    msgContext.getProperty(InboundWebsocketConstants.WEBSOCKET_SOURCE_HANDSHAKE_PRESENT).equals(true)) {
+            Integer errorCode = null;
+            String errorMessage = null;
+            if (msgContext.getProperty("errorCode") != null ) {
+                errorCode = Integer.parseInt(msgContext.getProperty("errorCode").toString());
+            }
+            if(msgContext.getProperty("ERROR_MESSAGE") != null ) {
+                errorMessage = msgContext.getProperty("ERROR_MESSAGE").toString();
+            }
+            try {
+                if (errorCode != null && errorMessage != null) {
+                    sourceHandler.handleClientWebsocketChannelTermination(
+                            new CloseWebSocketFrame(errorCode, errorMessage));
+                }
+            } catch (AxisFault fault) {
+                log.error("Error occurred while sending close frames", fault);
+            }
+            Object isConnectionAlive = ((Axis2MessageContext) msgContext).getAxis2MessageContext().getProperty
+                    (InboundWebsocketConstants.IS_CONNECTION_ALIVE);
+            if (isConnectionAlive != null && !(boolean) isConnectionAlive) {
+                InboundWebsocketChannelContext ctx = sourceHandler.getChannelHandlerContext();
+                ctx.writeToChannel(new CloseWebSocketFrame(1001, "shutdown"));
```

Comment:
This particular code section (from 78 to 84 lines) will be removed since IS_CONNECTION_ALIVE property was sent by the CGI TCP transport when an error occurred while TCP connection is persistent. Now the transport would send AxisFaults when an error occurred therefor this code snippet would not be in use.